var/spack/repos/builtin/packages/petsc/package.py

```diff
@@ -85,7 +85,7 @@ class Petsc(Package):
     depends_on('boost', when='@:3.5+boost')
     depends_on('metis@5:', when='+metis')
 
-    depends_on('hdf5+mpi', when='+hdf5+mpi')
```

Comment:
this is not even different from the original, before it will use `hdf5` if-and-only-if `petsc` is compiled with `mpi`. So it's not even about forwarding of variants. Same below.